src/bin/cargo.rs

```diff
@@ -91,7 +91,14 @@ macro_rules! each_subcommand{
         $mac!(verify_project);
         $mac!(version);
         $mac!(yank);
-    })
+    }
+}
+
+mod subcommands {
+    macro_rules! declare_mod {
+        ($name:ident) => ( pub mod $name; )
+    }
+    each_subcommand!(declare_mod);
```

Comment:
Weird, I'm kinda surprised that this doesn't require the path to the source file to be `src/bin/subcommands/foo.rs`...